lib/array-slice.js

```diff
@@ -40,16 +61,50 @@ var ArraySlice = createClass({
   },
 
   /**
-   * @param callback - Function to execute for each element
+   * Maps and then flattens the results.
+   * @param callback - Function to execute for each element.
+   * @param thisArg - Value to use as this (i.e the reference Object) when executing callback
+   * @returns {Element}
+   * @memberof ArraySlice.prototype
+   */
+  flatMap: function (callback, thisArg) {
+    var results = [];
+
+    this.forEach(function (element) {
+      var result = callback(element);
+
+      if (result) {
+        results.push(result);
+      }
+    }, this);
```

Comment:
The `this` here is probably redundant - the `forEach` callback does not access `this`, so it doesn't need to be changed.